docs/insomnia/context-object-reference.md

```diff
@@ -76,56 +80,117 @@ module.exports.responseHooks = [
 ];
 ```
 
+### Example: Alter a response body
+
+The response body works with [NodeJS Buffers](https://nodejs.org/api/buffer.html). To change the response body through a plugin, you'll need to translate to and from a Buffer.
+
+The below example ties into `responseHooks` and shows how to work with the NodeJS Buffers to:
+
+* Convert a Buffer to a string
+* Parse a string to a JS object
+* Modify the response by:
+  * Generating a random number
+  * Prompting to user for information in a modal
+* Convert the JS object to a string and then to a Buffer
+
+```js
+const bufferToJsonObj = buf => JSON.parse(buf.toString('utf-8'));
+const jsonObjToBuffer = obj => Buffer.from(JSON.stringify(obj), 'utf-8');
+
+module.exports.responseHooks = [
+  async ctx => {
+    try{
+      const resp = bufferToJsonObj(ctx.response.getBody());
+      
+      // Modify
+      resp.__randomNumber = Math.random();
+
+      // If you want something from a user, use a prompt:
+      const promptResult = await ctx.app.prompt('Type something', { defaultValue: 'abcd' });
+      resp.__customValue = promptResult;
+
+      ctx.response.setBody(jsonObjToBuffer(resp));
+    } catch {
+      // no-op
+    }
+  }
+]
+```
+_This example adds a `__randomNumber` and `__customValue` property to a JSON response. Update the functionality as needed._
+
 ## context.store
 Plugins can store persistent data via the storage context. Data is only accessible to the plugin that stored it.
 
-```
-// context.store functions
-async hasItem(key: string): Promise<boolean>
-async setItem(key: string, value: string): Promise<void>
-async getItem(key: string): Promise<string | null>
-async removeItem(key: string): Promise<void>
-async clear(): Promise<void>
-async all(): Promise<Array<{ key: string, value: string }>>
+```js
```

Comment:
I didn't realize they're all like this. I'll just let this one stay as a standin that we might as well update them all.
```suggestion
```ts
```